OmsAgent/omsagent.py

```diff
@@ -692,6 +735,51 @@ def get_vmresourceid_from_metadata():
         hutil_log_error('Unexpected error from Metadata service')
         return None
 
+def get_azure_region_from_imds():
+    req = urllib.request.Request(VMResourceIDMetadataEndpoint)
+    req.add_header('Metadata', 'True')
+
+    try:
+        response = json.loads(urllib.request.urlopen(req).read())
+
+        if ('compute' not in response or response['compute'] is None):
+            return None #classic vm
+
+        if ('location' not in response['compute'] or response['compute']['location'] is None):
+            return None #classic vm
+
+        return response['compute']['location']
+    except urllib.error.HTTPError as e:
+        hutil_log_error('Request to Metadata service URL ' \
+                        'failed with an HTTPError: {0}'.format(e))
+        hutil_log_info('Response from Metadata service: ' \
+                       '{0}'.format(e.read()))
+        return None
+    except:
+        hutil_log_error('Unexpected error from Metadata service')
+        return None
+
+def get_azure_cloud_domain():
+    try:
+        location = get_azure_region_from_imds()
+        # look for the cloud type associated with current location
+        for cloudName, data in CloudRegionMap.items():
+            domain = data['Domain']
+            regions = data['Regions']
+            for region in regions:
+                regionCompacted = region.replace(" ", "")
+                if regionCompacted.lower() == location.lower():
```

Comment:
`get_azure_region_from_imds()` can return `None`, in which case `location.lower()` will throw exception. We catch it, but better to handle `None` to start with IMO ...